### PR TITLE
Fix error handling in x509v3_cache_extensions and related functions

### PR DESCRIPTION
**Pull request by bernd-edlinger**
Like #10755 for 1.1.1 branch

### CODE REVIEW

File: crypto/x509v3/v3_purp.c [hunk 1]
```diff
@@ -324,16 +326,20 @@ static void setup_dp(X509 *x, DIST_POINT *dp)
...
 }
 
-static void setup_crldp(X509 *x)
+static int setup_crldp(X509 *x)
 {
     int i;
-    x->crldp = X509_get_ext_d2i(x, NID_crl_distribution_points, NULL, NULL);
-    for (i = 0; i < sk_DIST_POINT_num(x->crldp); i++)
-        setup_dp(x, sk_DIST_POINT_value(x->crldp, i));
+    x->crldp = X509_get_ext_d2i(x, NID_crl_distribution_points, &i, NULL);
```
Comment (FdaSilvaYY) [inline crypto/x509v3/v3_purp.c]: nit: blank line after declaration

Comment (bernd-edlinger) [reply crypto/x509v3/v3_purp.c]: Yes.

File: crypto/x509v3/v3_purp.c [hunk 2]
```diff
@@ -777,7 +811,11 @@ int X509_check_issued(X509 *issuer, X509 *subject)
         return X509_V_ERR_SUBJECT_ISSUER_MISMATCH;
 
     x509v3_cache_extensions(issuer);
+    if(issuer->ex_flags & EXFLAG_INVALID)
```
Comment (FdaSilvaYY) [inline crypto/x509v3/v3_purp.c]: Nit: miss these two nits in this version, sorry ;)

Comment (bernd-edlinger) [reply crypto/x509v3/v3_purp.c]: Thanks a lot!

**Comment by bernd-edlinger** [comment]
The travis failure, "The log length has exceeded the limit of 4 MB" looks unrelated.

**Comment by bernd-edlinger** [comment]
Merged to OpenSSL_1_1_1-stable ba4356a
Thanks!

**Pull request closed without merging**